packages/framework-core/src/booster-token-verifier.ts

```diff
@@ -121,6 +136,42 @@ export class BoosterTokenVerifier {
     )
     const winner = results.find((result) => result.status === 'fulfilled')
     if (winner) return Promise.resolve((winner as PromiseFulfilledResult<UserEnvelope>).value)
-    return Promise.reject(new Error(results.map((result) => (result as PromiseRejectedResult).reason).join('\n')))
+
+    return this.rejectVerification(results)
+  }
+
+  private rejectVerification(results: Array<PromiseSettledResult<UserEnvelope>>): Promise<UserEnvelope> {
+    const tokenExpiredErrors = this.getTokenExpiredErrors(results)
+    if (tokenExpiredErrors && tokenExpiredErrors.length > 0) {
+      const reasons = BoosterTokenVerifier.joinReasons(tokenExpiredErrors)
+      return Promise.reject(new BoosterTokenExpiredError(reasons))
+    }
+
+    const tokenNotBeforeErrors = this.getTokenNotBeforeErrors(results)
+    if (tokenNotBeforeErrors && tokenNotBeforeErrors.length > 0) {
+      const reasons = BoosterTokenVerifier.joinReasons(tokenNotBeforeErrors)
+      return Promise.reject(new BoosterTokenNotBeforeError(reasons))
+    }
+
+    const reasons = BoosterTokenVerifier.joinReasons(results as Array<PromiseRejectedResult>)
+    return Promise.reject(new NotAuthorizedError(reasons))
```

Comment:
One thing to consider: If the token includes a `kid` field, it could make sense to return the error from the validator that matches that specific `kid`, ignoring the others. It could also make sense to run only that specific validator instead of all of the existing ones in parallel.